x-pack/plugins/fleet/server/services/agents/helpers.ts

```diff
@@ -29,7 +33,11 @@ export function searchHitToAgent(hit: FleetServerAgentESResponse & { sort?: Sort
     sort: hit.sort,
   };
 
-  agent.status = getAgentStatus(agent);
+  if (!hit.fields?.status?.length) {
```

Comment:
@juliaElastic I think this may be overkill, I am protecting against someone calling this method with a doc that does not have the runtime field, btu I am worried some unpredicted production situation would break fleet here.

Should I just log a warning and default the status to online? I think omitting the agent status is not an option here as it will have too many follow on effects

Reply:
I agree that it would be better to log the error and not throw if this is not a critical info. We have seen many SDHs where Fleet was broken due to a non-critical error.